### PR TITLE
Implementation of the linked-list exercise 

### PR DESCRIPTION
**Pull request by jspengeman**
Looking for some feedback on my migration of the ecmascript linked-list exercise #71.

### CODE REVIEW

**Comment by jspengeman** [comment]
Looks like I borked the source history. I can fix it by cherry picking my commits and moving onto a different branch if people would like prior to reviewing the PR?

File: exercises/linked-list/linked-list.test.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import LinkedList from './linked-list'
+
+describe('LinkedList', () => {
+  test('add/extract elements to the end of the list with push/pop', () => {
+    const list = new LinkedList<number>()
+    list.push(10)
+    list.push(20)
+    expect(list.pop()).toBe(20)
+    expect(list.pop()).toBe(10)
+  })
```
[the remark this answers is not in the document]
Comment (jspengeman) [reply exercises/linked-list/linked-list.test.ts]: Good bot.

**Comment by masters3d** [comment]
Do you mind fixing the merge conflict?

**Comment by jspengeman** [comment]
@masters3d no problem! All fixed up now.

**Comment by masters3d** [comment]
THanks!

**Comment by jspengeman** [comment]
No problem! Thank you. Enjoyed working on this. I will probably pick up another exercise.

**Pull request merged**